skorch/dataset.py

```diff
@@ -75,6 +89,41 @@ def uses_placeholder_y(ds):
     return isinstance(ds, Dataset) and hasattr(ds, "y") and ds.y is None
 
 
+ERROR_MSG_1_ITEM = (
+    "You are using a non-skorch dataset that returns 1 value. "
+    "Remember that for skorch, Dataset.__getitem__ must return exactly "
+    "2 values, X and y (more info: "
+    "https://skorch.readthedocs.io/en/stable/user/dataset.html).")
+
+
+ERROR_MSG_MORE_THAN_2_ITEMS = (
+    "You are using a non-skorch dataset that returns {} values. "
+    "Remember that for skorch, Dataset.__getitem__ must return exactly "
+    "2 values, X and y (more info: "
+    "https://skorch.readthedocs.io/en/stable/user/dataset.html).")
+
```

Comment:
The error messages are not exactly the same, so they are not duplicates. Or do you suggest to define the common part separately, or merge the two and format the distinct part?

One of the reasons I chose to define the messages as such was to minimize the overhead due to creating/formatting the string during the `unpack_data` call because it could be called very often. But one could argue that this formatting would only when there is an error, in which case it is only called once. Idk if there are cases where the error would be triggered repeatedly and thus cause overhead.

Reply:
Lines 23-34 are the same as lines 92-103.